notifier.rst

```diff
@@ -0,0 +1,562 @@
+.. index::
+   single: Notifier
+
+Sending Notifications with Notifier
+===================================
+
+.. versionadded:: 5.0
+
+    The Notifier component was introduced in Symfony 5.0 as an
+    :doc:`experimental feature </contributing/code/experimental>`.
+
+Current web applications use many different channels to send messages to
+the users (e.g. SMS, Slack messages, emails, push notifications, etc.). The
+Notifier component in Symfony is an abstraction on top of all these
+channels. It provides a dynamic way to manage how the messages are send.
+Get the Notifier installed using:
+
+.. code-block:: terminal
+
+    $ composer require symfony/notifier
+
+Channels: Chatters, Texters, Email and Browser
+----------------------------------------------
+
+The notifier component can send notifications to different channels. Each
+channel can integrate with different providers (e.g. Slack or Twilio SMS)
+by using transports.
+
+The notifier component supports the following channels:
+
+* `SMS <SMS Channel>`_ for simple text based notifications;
+* `Chat <Chat Channel>`_ for advanced text notifications;
+* `Email <Email Channel>`_ integrating the :doc:`Symfony Mailer </mailer>`;
+* Browser using :ref:`flash messages <flash-messages>`.
+
+.. tip::
+
+    Use :doc:`secrets </configuration/secrets>` to securily store your
+    API's tokens.
+
+.. _notifier-texter-dsn:
+
+SMS Channel
+~~~~~~~~~~~
+
+The SMS channel uses :class:`Symfony\\Component\\Notifier\\Texter` classes
+to send plain text messages to a user. Symfony provides integration with a
+couple popular SMS services:
+
+=======  ===========================  ========================================
+Service  Package                      DSN
+=======  ===========================  ========================================
+Twilio   ``symfony/twilio-notifier``  ``twilio://SID:TOKEN@default?from=FROM``
+Nexmo    ``symfony/nexmo-notifier``   ``nexmo://KEY:SECRET@default?from=FROM``
```

Comment:
personal question, i am using a feature given by one of the french phone provider (Free Mobile)
to notify/alert me on some of my project
If i want to integrates it via the sf/notifier, should i create a new symfony/TODO-notifier?

Reply:
No, you don't have to create a package. You'll have to create a TransportFactory (by extending https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Notifier/Transport/AbstractTransportFactory.php ) and a Transport (by extending https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Notifier/Transport/AbstractTransport.php ).

And then make sure to tag the transport factory as `chatter.transport_factory` in the services config.

Reply:
Will give it a try